server/index.js

```diff
@@ -17,8 +18,54 @@ const OIDC_USE_PKCE = process.env.OIDC_USE_PKCE === "true" || false;
 const OIDC_METADATA = JSON.parse(process.env.OIDC_METADATA || '{}');
 const clientMetadata = Object.assign({client_id: OIDC_CLIENT_ID, client_secret: OIDC_SECRET}, OIDC_METADATA);
 
+/*
+    Code copied from https://stackoverflow.com/questions/63309409/creating-a-code-verifier-and-challenge-for-pkce-auth-on-spotify-api-in-reactjs
```

Comment:
Is there a reason why we move it into the index file instead of a separate file?

Reply:
There's no easy way to import unless I convert this app to use a node.js server (currently express.js)

Reply:
Which will be a large refactor..